test/fakes/fakeupstreamauthority/fakeupstreamauthority.go

```diff
@@ -45,7 +47,7 @@ type UpstreamAuthority struct {
 	config     Config
 }
 
-func New(t *testing.T, config Config) *UpstreamAuthority {
+func New(t *testing.T, config Config) *catalog.UpstreamAuthority {
```

Comment:
I think you can revert these changes. The established pattern is to have the fakeservercatalog package provide a helper for constructing a `catalog.UpstreamAuthority` from a `name` and an `upstreamauthority.UpstreamAuthority`. See the following function:
https://github.com/spiffe/spire/blob/4ccc31d85e4d1775c1f0f3d6ee9d07bdc0c9da8a/test/fakes/fakeservercatalog/catalog.go#L55

This makes it easy to use any `upstreamauthority.UpstreamAuthority` implementation without requiring that it also implement `catalog.PluginInfo`.